src/sprout/Helpers/AI/WorkerAiContentProcess.php

```diff
@@ -0,0 +1,154 @@
+                switch ($item['activation_status']) {
+                    case 'active':
+                        $active_fields['active'] = 1;
+                        $active_msg = "Setting record #{$item['target_id']} in {$item['target_table']} to ACTIVE";
+                        break;
+                    case 'inactive':
+                        $active_fields['active'] = 0;
+                        $active_msg = "Setting record #{$item['target_id']} in {$item['target_table']} to INACTIVE";
+                        break;
+                    default:
+                        // Make no
+                        break;
+                }
+            }
+
+            try {
+                // Fire the actual AI class method to get the content
+                $output = $class::$method($item['prompt']);
+
+                // Merge the active flag change with the AI content
+                $data = array_merge([$item['target_col'] => $output], $active_fields);
+                Pdb::update($item['target_table'], $data, ['id' => $item['target_id']]);
+
+                Pdb::update('ai_content_queue', ['status' => 'complete'], ['id' => $item['id']]);
+
+                Worker::message("Updated content for record #{$item['target_id']} in {$item['target_table']}");
+                if ($active_msg) Worker::message($active_msg);
+
+                $success++;
+
+                $cost = $class->getLastRequestCost();
+                $unit = $class->getRequestCostUnit();
+                Worker::message("Request cost: {$cost} {$unit}");
+
+                if (!isset($total_costs[$unit])) $total_costs[$unit] = 0;
+                $total_costs[$unit] += $cost;
+
+            } catch (Exception $e) {
+                // We will want this in the logs in case our integration has broken
+                Kohana::logException($e);
+
+                $log = $e->getMessage();
+                Pdb::update('ai_content_queue', ['status' => 'failed', 'log' => $log], ['id' => $item['id']]);
+
+                Worker::message("AI FAILED for record #{$item['target_id']} in {$item['target_table']}: {$log}");
+                $failed++;
+            }
+
+            if (!$transacting) Pdb::commit();
+
+            // Line break in output
+            Worker::message('');
+
+            // Avoid continuously cycling an error
+            $ids[] = $item['id'];
+
+            $conditions = $params = [];
+            $conditions[] = ['status', 'IN', $status];
+            $conditions[] = ['id', 'NOT IN', $ids];
+
+            $where = Pdb::buildClause($conditions, $params);
+            $q_items = "SELECT * FROM ~ai_content_queue WHERE {$where} ORDER BY id ASC LIMIT 1";
+            $items = Pdb::query($q_items, $params, 'arr');
+        }
+
+        Worker::message('AI Content Processed: ' . $success . ' success, ' . $failed . ' failed');
+
+        foreach ($total_costs as $unit => $cost) {
+            Worker::message("Total {$unit} cost: {$cost}");
+        }
```

Comment:
We're not storing these costs anywhere for consumption?

Can non-super users read the worker log?